### PR TITLE
Better error messaging when infer_shape fails

### PR DESCRIPTION
**Pull request by leopd**
Previously, when infer_shape failed, it would show the difference between the provided and inferred shapes for gradients, gradient-arguments, gradient-values, or aux-states, but NOT plain arguments.  I fixed that so it shows the full error message for normal arguments too.

So previously you'd just see:
```
mxnet.base.MXNetError: InferShape Error in loss's label argument
```

Now it shows:
```
mxnet.base.MXNetError: InferShape Error in loss's label argument
Shape inconsistent, Provided=(50,1024), inferred shape=(50,100)
```

(Also just removed an extra space in the "shape inconsistent" message to make it display a bit more cleanly.)

Also, I added a bit of python code to print the arguments passed in to infer_shape() in case of failure -- to further help with debugging infer_shape problems.

### CODE REVIEW

File: python/mxnet/symbol.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 from .base import _LIB
 from .base import c_array, c_str, mx_uint, py_str, string_types, mx_real_t
 from .base import NDArrayHandle, ExecutorHandle, SymbolHandle
-from .base import check_call, ctypes2docstring
+from .base import check_call, ctypes2docstring, MXNetError
```
Comment (piiswrong) [inline python/mxnet/symbol.py]: The mxneterror thing doesn't seem very infromative. How about use appropriate default errors?

Comment (leopd) [inline python/mxnet/symbol.py]: As I understand it, `MXNetError` is the wrapper class for all exceptions thrown in C++ code when they surface in python code.  I'm not sure how to change that.

**Comment by piiswrong** [comment]
plz fix lint:
```
03:06:09  python/mxnet/symbol.py:462: convention (C0326, bad-whitespace, ) Exactly one space required after comma
03:06:09              for i,arg in enumerate(args):
03:06:09                   ^
03:06:09  python/mxnet/symbol.py:463: convention (C0326, bad-whitespace, ) Exactly one space required after comma
03:06:09                  print("  #%d: %s" % (i,arg))
03:06:09                                        ^
03:06:09  python/mxnet/symbol.py:464: convention (C0326, bad-whitespace, ) Exactly one space required after comma
03:06:09              for k,v in kwargs.items():
03:06:09                   ^
03:06:09  python/mxnet/symbol.py:465: convention (C0326, bad-whitespace, ) Exactly one space required after comma
03:06:09                  print("  %s: %s" % (k,v))
```

**Comment by leopd** [comment]
Sure thing.  Thanks for pasting those in, Eric.  I need to get better at reading the lint output more closely.

**Pull request merged**